### PR TITLE
Replace no_backup_and_restore with no_restore

### LINKED ISSUE #27178: Support sub-app restart with different sub-app configuration
### Motivation

In reactor physics, it is common to perform an accident analysis for a depleted core condition. This is achieved by depleting a core to a certain point and restarting a transient calculation from it. Depletion calculation is a pseudo-state-state calculation, so sub-apps are typically configured in the steady-state mode (`Steady` executioner and `FullSolveMultiApp`). However, restarting a transient calculation from it requires changing sub-app configurations. Under this condition, it seems that the main-app cannot restart sub-apps automatically.

### Design

Support changing sub-app configurations during restart.

### Impact

Checkpoint restart will be able to support more variety of applications.

### LINKED ISSUE #27223: Do not disable backups in `FullSolveMultiApp`
### Motivation

The parameter `no_backup_and_restore` in `FullSolveMultiApp` causes issues when you actually need to have something backed up at the end of the solve. For example, you have a steady state solve with `Outputs/checkpoint=true` and want to restart that state into a transient, even though you don't want to use the state within inner iterations.

 Or, if you're doing picard within timesteps, your solve is killed, and you want to recover at the end of that timestep on a new run.

### Design

Replace `no_backup_and_restore` with `no_restore`. Keep the behavior same where you don't do the restore, but always do the backup.

### Impact

Minor parameter change. Will require changing a few tests, but the behavior should be the same.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #27223 and closes #27178

I'll see if I can come up with a test. All previous behavior should still work as the lack-of-restoration was the important part to disable.

### CODE REVIEW

**Comment by loganharbour** [comment]
@grmnptr need your input on the stochastic tools changes. I think that this still satisfies the required behavior, but a sanity check would be great.

**Comment by YaqiWang** [comment]
Looks good to me.

**Comment by NamjaeChoi** [comment]
Confirmed that it works as intended.

File: framework/src/multiapps/FullSolveMultiApp.C [hunk 1]
```diff
@@ -23,12 +23,17 @@ FullSolveMultiApp::validParams()
...
   InputParameters params = MultiApp::validParams();
   params.addClassDescription("Performs a complete simulation during each execution.");
-  params.addParam<bool>(
+  params.addDeprecatedParam<bool>(
       "no_backup_and_restore",
       false,
-      "True to turn off backup/restore for this multiapp. This is useful when doing steady-state "
+      "True to turn off restore for this multiapp; use 'no_restore' instead.",
+      "This functionality has been replaced with 'no_restore' and now only disables restoration.");
+  params.addParam<bool>(
```
Comment (GiudGiud) [inline framework/src/multiapps/FullSolveMultiApp.C:31]: use params.addParam params.deprecateParam instead (see ParsedAux for an example)

then you can remove the first check in the constructor

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**